### DISCUSSION TITLE
Understand the CellReads.stat file output from STARsolo

### DISCUSSION THREAD

**Discussion by CassandreHeritier** [alexdobin/STAR#1826] [Q&A] [answered]
Hello,

I want to understand the output file `Solo.out/GeneFull_Ex50pAS/CellReads.stats` from STARsolo with options `--soloCellReadStats Standard` and `--soloFeatures GeneFull_Ex50pAS` but I don't find any manual or instruction on the signification of each column, neither in the STAR manual.

The columns are: `CB cbMatch cbPerfect cbMMunique cbMMmultiple genomeU genomeM featureU featureM exonic intronic exonicAS intronicAS mito countedU countedM nUMIunique nGenesUnique nUMImulti nGenesMulti`

And the file looks like:

![image](https://user-images.githubusercontent.com/72193398/231377768-2d39006f-7779-48bf-83e1-3dbf37ad7ac0.png)

I want to know for each read if the read is intronic or exonic type.

Thanks!

**Comment by alexdobin** [comment, marked as answer]
Hi @CassandreHeritier 

This file shows the mapping statistics of reads in each cell.
There is a discussion about these fields here:
https://github.com/alexdobin/STAR/issues/1501#issue-1161546794
Here is the summary:

`cbMatch` means the number of read that matched the cell barcode,
`cbPerfect` for the number of perfect match on cell barcode
```cbMMunique``` : number of reads with cell barcodes that map with mismatches to one barcode in the passlist
```cbMMmultiple``` : number of reads with cell barcodes that map with mismatches to multiple barcodes in the passlist

```genomeU``` : number of reads mapping to one locus in the genome
```genomeM``` : number of reads mapping to multiple loci in the genome
```featureU``` : number of reads mapping to one feature (Gene, GeneFull, etc)
```featureM``` : number of reads mapping to multiple features

```exonic```: number of reads mapping to annotated exons
```intronic```: number of reads mapping to annotated introns
These are only calculated for ```--soloFeatures GeneFull_Ex50pAS``` and/or ```GeneFull_ExonOverIntron```

```exonicAS``` : number of reads mapping antisense to annotated exons
```intronicAS``` : number of reads mapping antisense to annotated introns
These are only calculated for ```--soloFeatures GeneFull_Ex50pAS```

`countedU`: number of unique-gene reads whose UMIs contributed to counts in the matrix.mtx
`countedM`: number of multi-gene reads whose UMIs contributed to counts in the matrix.mtx

`mito` for the number of read mapping on mitochondrial genome


`nUMIunique` for the total number of counted UMI for unique-gene reads
`nGenesUnique` for the number of genes for unique-gene reads
`nUMImulti` for the total number of counted UMI for multi-gene reads
`nGenesMulti` for the number of genes for multi-gene reads

**Reply by CassandreHeritier** [reply]
Thank you so much :100: :+1:

**Reply by huiz0916** [reply]
> Hi @alexdobin 
> 
> This file shows the mapping statistics of reads in each cell. There is a discussion about these fields here: [#1501 (comment)](https://github.com/alexdobin/STAR/issues/1501#issue-1161546794) Here is the summary:
> 
> `cbMatch` means the number of read that matched the cell barcode, `cbPerfect` for the number of perfect match on cell barcode `cbMMunique` : number of reads with cell barcodes that map with mismatches to one barcode in the passlist `cbMMmultiple` : number of reads with cell barcodes that map with mismatches to multiple barcodes in the passlist
> 
> `genomeU` : number of reads mapping to one locus in the genome `genomeM` : number of reads mapping to multiple loci in the genome `featureU` : number of reads mapping to one feature (Gene, GeneFull, etc) `featureM` : number of reads mapping to multiple features
> 
> `exonic`: number of reads mapping to annotated exons `intronic`: number of reads mapping to annotated introns These are only calculated for `--soloFeatures GeneFull_Ex50pAS` and/or `GeneFull_ExonOverIntron`
> 
> `exonicAS` : number of reads mapping antisense to annotated exons `intronicAS` : number of reads mapping antisense to annotated introns These are only calculated for `--soloFeatures GeneFull_Ex50pAS`
> 
> `countedU`: number of unique-gene reads whose UMIs contributed to counts in the matrix.mtx `countedM`: number of multi-gene reads whose UMIs contributed to counts in the matrix.mtx
> 
> `mito` for the number of read mapping on mitochondrial genome
> 
> `nUMIunique` for the total number of counted UMI for unique-gene reads `nGenesUnique` for the number of genes for unique-gene reads `nUMImulti` for the total number of counted UMI for multi-gene reads `nGenesMulti` for the number of genes for multi-gene reads

Thanks for explanations. `nUMIunique` means all the umi without de-duplicate or has removed duplication?
